### PR TITLE
Initial implementation of the fills color panel

### PR DESCRIPTION
**Pull request by Alecaddd**
## Summary / How this PR fixes the problem?
Very simple initial PR to restore the fills color controls in the UI by using the same virtualizing listbox used in the layers panel.

## Known Issues / Things To Do
- [x] Convert the `Components.Fill` into a simple `struct`.
- [x] Create the repeatable UI of the color row.
- [x] Defer the creation of heavy widgets only when needed.

### CODE REVIEW

**Comment by Alecaddd** [comment]
This is ready for a review since it re-implements a lot of things.
The major thing missing is the color chooser when clicking on the color button. I'll do that later since it needs to be converted to a utility widget that can be reused.

**Comment by Alecaddd** [comment]
Converted to a draft due to some substantial architectural changes to this whole area.

File: src/Models/ColorModel.vala [hunk 1]
```diff
@@ -21,72 +21,74 @@
...
-    private unowned Lib.Components.Fill? fill;
-    private unowned Lib.Components.Border? border;
+    public signal void value_changed ();
 
-    public Type type;
-    public enum Type {
-        FILL,
-        BORDER
-    }
+    public class SignalBlocker {
```
Comment (mbfraga) [inline src/Models/ColorModel.vala:29]: Why does this signal blocker exist?

File: src/Lib/Components/Fills.vala [hunk 2]
```diff
@@ -58,20 +106,50 @@ public class Akira.Lib.Components.Fills : Component, Copyable<Fills> {
...
         }
-        return tmp;
+        return false;
     }
 
-    public void prep_fills (uint number_to_prep) {
-        data = new Fill.FillData[number_to_prep];
-        for (var i = 0; i < number_to_prep; ++i) {
-            data[i]._id = i;
+    public void append (Fill fill) {
```
Comment (mbfraga) [inline src/Lib/Components/Fills.vala:129]: don't add public mutators to components. This breaks the model contract and thus copy/paste/undo/redo

**Review by mbfraga — CHANGES_REQUESTED** [review]

File: src/Models/ColorModel.vala (hunk 1 above)
Comment (Alecaddd) [reply src/Models/ColorModel.vala:29]: It's used by the FillitemModel when the node is updated for the first time, which blocks the `on_value_changed()` in the subclass.
Should the signal blocker live only in the subclass?

File: src/Lib/Components/Fills.vala (hunk 2 above)
Comment (Alecaddd) [reply src/Lib/Components/Fills.vala:129]: What would you recommend to do here to handle multiple single `Fill` for the same `Fills` component?

**Pull request merged**